receiver/hostmetricsreceiver/internal/scraper/cpuscraper/cpu_scraper.go

```diff
@@ -64,31 +69,21 @@ func (s *scraper) scrape(_ context.Context) (pdata.Metrics, error) {
 		return md, scrapererror.NewPartialScrapeError(err, metricsLen)
 	}
 
-	initializeCPUTimeMetric(metrics.AppendEmpty(), s.startTime, now, cpuTimes)
-	return md, nil
-}
-
-func initializeCPUTimeMetric(metric pdata.Metric, startTime, now pdata.Timestamp, cpuTimes []cpu.TimesStat) {
-	metadata.Metrics.SystemCPUTime.Init(metric)
-
-	ddps := metric.Sum().DataPoints()
-	ddps.EnsureCapacity(len(cpuTimes) * cpuStatesLen)
 	for _, cpuTime := range cpuTimes {
-		appendCPUTimeStateDataPoints(ddps, startTime, now, cpuTime)
+		recordCPUTimeStateDataPoints(s.mb, now, cpuTime)
 	}
+	s.mb.Emit(metrics)
+	return md, nil
 }
 
 const gopsCPUTotal string = "cpu-total"
 
-func initializeCPUTimeDataPoint(dataPoint pdata.NumberDataPoint, startTime, now pdata.Timestamp, cpuLabel string, stateLabel string, value float64) {
-	attributes := dataPoint.Attributes()
+func recordCPUTimeDataPoint(mb *metadata.MetricsBuilder, now pdata.Timestamp, cpuAttribute string,
+	stateLabel string, value float64) {
```

Comment:
Maybe just personal preference, but I would encourage that the metrics builder is not passed around, that instead this is a receiver function:

```suggestion
func (s *scraper) recordCPUTimeDataPoint(now pdata.Timestamp, cpuAttribute string, stateLabel string, value float64)
	...
	s.mb.RecordSystemCPUTimeDataPoint(...)
```

What I like about this pattern is that the parameters to the function are purely descriptive of the data point, and the concern of "how to build the data point" is assigned to the scraper.

Ultimately, this is just an individual implementation detail, but we are encouraging a pattern to reuse many times, so I think it's worth considering here.

Reply:
Agree, makes sense to establish this pattern. I'll change it